crates/ide_completion/src/item.rs

```diff
@@ -152,9 +144,34 @@ pub struct Relevance {
     pub exact_type_match: bool,
 }
 
-impl Relevance {
+impl CompletionRelevance {
+    /// Provides a relevance score. Higher values are more relevant.
+    ///
+    /// The absolute value of the relevance score is not meaningful, for
+    /// example a value of 0 doesn't mean "not relevant", rather
+    /// it means "least relevant". The score value should only be used
+    /// for relative ordering.
+    ///
+    /// See is_relevant if you need to make some judgement about score
+    /// in an absolute sense.
+    pub fn score(&self) -> u8 {
```

Comment:
No strong opinion here, but I tend to use `u32` for "don't really care about the range" types. working with u8, on the instruction level, needs some extra bit manipulations.

Reply:
Good to know we are okay using four bytes here. clangd [uses an f32](https://github.com/llvm/llvm-project/blob/bc4a5bdce4af82a5522869d8a154e9e15cf809df/clang-tools-extra/clangd/Quality.cpp#L403) which seems like it could be something we'd want to do in the future. 

The primary downside of a u32 vs a u8 is that it makes the actual `sortText` string larger. Fortunately we can do nearly all of our testing against relevance rather than sort_text, so this doesn't matter too much. 

I went ahead and changed this to u32.